configuration/contrib/log-plugins/dmesg

```diff
@@ -0,0 +1,164 @@
+#!/bin/sh
+set -eu
+
+NOT_SUPPORTED=2
+
+usage() {
+    cat <<EOT
+dmesg log plugin for thin-edge.io
+
+IMPORTANT: Some dmesg implementations don't support date/time filters, and log facility
+
+The log plugin allows users to access kernel messages from dmesg. It offers multiple types based
+on the dmesg log "facility" and a general "all" type to allow retrieval of all messages regardless
+of the facility.
+
+Usage:
+  $0 list
+      List all log types supported by this plugin
+
+  $0 get <log-type> [options]
+      Fetch logs for the given facility (<log-type>)
+
+Options for "get":
+  --since <timestamp>   Show logs since the given timestamp (seconds since epoch)
+  --until <timestamp>   Show logs up to the given timestamp  (seconds since epoch)
+  --help                Show this help message and exit
+
+Examples:
+  $0 list
+  $0 get all
+
+  # Note: busybox implementations of dmesg may not support the facility filtering
+  $0 get daemon --since 1696250000 --until 1696260000
+EOT
+}
+
+is_busy_box() {
+    if dmesg --help 2>&1 | grep -q "facility"; then
+        return 1
+    fi
+    return 0
+}
+
+list_log_types() {
+    if ! command -V dmesg >/dev/null 2>&1; then
+        exit "$NOT_SUPPORTED"
+    fi
+
+    printf 'all\n'      # No facility filtering
+
+    # add optional facility filtering if the version of dmesg supports it
+    # as busybox implementations don't seem to support it
+    if ! is_busy_box; then
+        printf 'kern\n'     # kernel messages
+        printf 'user\n'     # random user-level messages
+        printf 'mail\n'     # mail system
+        printf 'daemon\n'   # system daemons
+        printf 'auth\n'     # security/authorization messages
+        printf 'syslog\n'   # messages generated internally by syslogd
+        printf 'lpr\n'      # line printer subsystem
+        printf 'news\n'     # network news subsystem
```

Comment:
The standard set seems to be 15: https://techdocs.broadcom.com/us/en/symantec-security-software/web-and-network-security/security-analytics/8-2-6/_reference_home/syslog.html

Reply:
Though I'll still have to add a check to find out when the datetime filtering can be used (as that is also dependent on the implementation).

Reply:
Though the problem is that I don't think that all dmesg implementations support it anyway...so it would just be adding an extra 15 items to the types, where we don't know which ones would work or not.

So if we keep things simple for now, and just have one type, as it is anyway for more advanced usecases (as this is more for powerusers who want to monitor kernel messages to debug driver problems etc.)